server/src/api/chat_completions.rs

```diff
@@ -0,0 +1,336 @@
+//! Responsible for creating the json schema associated with the AtomaAPI, which is modeled after OpenAPI.
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Deserializer, Serialize};
+use serde_json::Value;
+
+// NOTE: `Role`, `Message` and `content` in some of the `Message`s have sort of a strange
+// relationship, ie. `content` is sometimes a struct, sometimes a string or an array, and
+// `Message` is expected to be a struct, but is defined as if it should be an enum even though
+// the `Role` is defining the type of `Message`. This may need some clarification either via
+// documentation to distinguish the differences between the OpenAPI and AtomaAPI, or further
+// work to make them the same.
+
+/// ID of the model to use.
+#[derive(Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(rename(serialize = "model", deserialize = "model"))]
+pub enum Model {
+    #[serde(rename(serialize = "llama3", deserialize = "llama3"))]
+    Llama3,
```

Comment:
We will include many different models, including different versions of the most recent Llama release (llama3.1 8b, 70b and 405b) and even previous version of Llama.

Reply:
Right, this is just a stub, I think I have a TODO somewhere referencing this.